biothings/web/analytics/channels.py

```diff
@@ -1,83 +1,99 @@
 
 
 class SlackChannel(Channel):
     def __init__(self, hook_urls):
         self.hooks = hook_urls
 
-    def handles(self, event):
+    async def handles(self, event):
         return isinstance(event, Message)
 
-    def send(self, message):
-        for url in self.hooks:
-            yield HTTPRequest(
-                url=url,
-                method="POST",
-                headers={"content-type": "application/json"},
-                body=orjson.dumps(message.to_slack_payload()).decode(),
-                ca_certs=certifi.where(),  # for Windows compatibility
-            )
-
+    async def send(self, event):
+        async with aiohttp.ClientSession() as session:
+            tasks = [self.send_request(session, url, event) for url in self.hooks]
+            await asyncio.gather(*tasks)
 
-# Measurement Protocol (Universal Analytics)
-# https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide
+    async def send_request(self, session, url, event):
+        async with session.post(
+                url,
+                json=event.to_slack_payload(),
+                verify=certifi.where()  # for Windows compatibility
+        ) as _:
+            pass
 
 
 class GAChannel(Channel):
     def __init__(self, tracking_id, uid_version=1):
         self.tracking_id = tracking_id
         self.uid_version = uid_version
 
-    def handles(self, event):
+    async def handles(self, event):
         return isinstance(event, Event)
 
-    def send(self, payload):
+    async def send(self, payload):
         events = payload.to_GA_payload(self.tracking_id, self.uid_version)
-        # #batch-limitations section of the URL above
-        # A maximum of 20 hits can be specified per request.
-        for i in range(0, len(events), 20):
-            yield HTTPRequest(
-                "http://www.google-analytics.com/batch",
-                method="POST",
-                body="\n".join(events[i : i + 20]),  # noqa: E203
-            )
+        async with aiohttp.ClientSession() as session:
+            for i in range(0, len(events), 20):
+                data = "\n".join(events[i:i + 20])
+                url = "http://www.google-analytics.com/batch"
+                await self.send_request(session, url, data)
+
+    async def send_request(self, session, url, data):
+        async with session.post(url, data=data) as _:
+            pass
 
 
 class GA4Channel(Channel):
     def __init__(self, measurement_id, api_secret, uid_version=1):
         self.measurement_id = measurement_id
         self.api_secret = api_secret
         self.uid_version = uid_version
+        self.max_retries = 1
 
-    def handles(self, event):
+    async def handles(self, event):
         return isinstance(event, Event)
 
-    def send(self, payload):
-        """
-
-        Limitations:
-        https://developers.google.com/analytics/devguides/collection/protocol/ga4/sending-events?client_type=gtag
-        """
+    async def send(self, payload):
```

Comment:
`arguments-renamed: Parameter 'event' has been renamed to 'payload' in overriding 'GA4Channel.send' method`. Not essential just highlighting the API change for different channels. This existed before your PR as well

Reply:
renamed all of them to `event`